core/autoscaling.go

```diff
@@ -20,16 +27,149 @@ type autoScalingGroup struct {
 
 	// spot instance requests generated for the current group
 	spotInstanceRequests []*spotInstanceRequest
+	minOnDemand          int64
 }
 
-func (a *autoScalingGroup) process() {
+func (a *autoScalingGroup) loadPercentageOnDemand(tagValue *string) (int64, bool) {
+	percentage, err := strconv.ParseFloat(*tagValue, 64)
+	if err != nil {
+		logger.Printf("Error with ParseFloat: %s\n", err.Error())
+	} else if percentage == 0 {
+		logger.Printf("Loaded MinOnDemand value to %f from tag %s\n", percentage, OnDemandPercentageLong)
+		return int64(percentage), true
+	} else if percentage > 0 && percentage <= 100 {
+		instanceNumber := float64(len(a.instances.catalog))
+		onDemand := int64(math.Floor((instanceNumber * percentage / 100.0) + .5))
```

Comment:
I'll comment just in case of someone else reading it. Basically, if you do no add a value (here 0.5), then Floor will always return 0, when the value is below 1. So in our tests with 3 instances 33% (which is in fact 0.333333333) will return 0.99 of an instance, and therefore Floor would return 0 instead of 1.

So the value 0.5 says, that if a percentage return more than 0.5 of an instance, to round that up to 1. I could add extra tests to make this more explicit, in a situation with 3 instances, 16% would be ~0.49 so would return 0 instance, but 17% or (16.66666667%) would return 1 instance, as it would be rounded up.

More information can be found here if necessary: https://github.com/golang/go/issues/4594